libbpf-rs/src/object.rs

```diff
@@ -183,40 +198,135 @@ impl Object {
         }
     }
 
-    pub fn map<T: AsRef<str>>(&mut self, name: T) -> Result<Option<&mut MapBuilder>> {
+    pub fn map<T: AsRef<str>>(&mut self, name: T) -> Result<Option<&mut MapModifier>> {
         if self.maps.contains_key(name.as_ref()) {
             Ok(self.maps.get_mut(name.as_ref()))
+        } else if let Some(ptr) = find_map_in_object(self.ptr, name.as_ref())? {
+            self.maps
+                .insert(name.as_ref().to_owned(), MapModifier::new(ptr));
+            Ok(self.maps.get_mut(name.as_ref()))
         } else {
-            let c_name = util::str_to_cstring(name.as_ref())?;
-            let ptr =
-                unsafe { libbpf_sys::bpf_object__find_map_by_name(self.ptr, c_name.as_ptr()) };
-            if ptr.is_null() {
-                Ok(None)
+            Ok(None)
+        }
+    }
+
+    // Same as [`map`] except will panic if `Err` or `None` is encountered.
+    pub fn map_unwrap<T: AsRef<str>>(&mut self, name: T) -> &mut MapModifier {
+        self.map(name).unwrap().unwrap()
+    }
+
+    pub fn prog<T: AsRef<str>>(&mut self, name: T) -> Result<Option<&mut ProgramModifier>> {
```

Comment:
For prog, lookup by name is a bit more ambiguous than for map. Maps only have name, which correspond to their respective variable name in C code.

But programs actually have section name, associated with them (which is what libbpf calls "title", see bpf_object__find_program_by_title) and then the name of C function (that's what bpf_object__find_program_by_name uses). Should we have both look up methods?

Name is probably good enough for starters... Don't have strong opinion. In libbpf/C land, historically most uses are lookups by title (section name), but I think overall lookup by name is better in the long run (I think eventually libbpf will allow to have multiple independent functions with the same section name). So just consider what would you do if someone requested lookup by section name.